src/results.rs

```diff
@@ -0,0 +1,278 @@
+#[pyclass]
+pub struct RowsResult {
+    row_col_iterator: Py<RowColumnIterator>,
+    factory: Py<RowFactory>,
+}
+
+#[pymethods]
+impl RowsResult {
+    pub fn __next__(&mut self) -> PyResult<Py<PyAny>> {
+        Python::with_gil(|py| {
+            self.row_col_iterator
+                .borrow_mut(py)
+                .yoked
+                .with_mut_return(|view| view.next_row())?;
+
+            self.factory
+                .call_method1(py, "build", (&self.row_col_iterator.bind(py),))
+        })
+    }
+
+    pub fn __iter__(slf: PyRefMut<'_, Self>) -> PyRefMut<'_, Self> {
+        slf
+    }
+}
+
+#[derive(Yokeable)]
+struct IteratorWrapper<'a> {
+    row_iterator: RawRowIterator<'a, 'a>,
+    column_iterator: ColumnIterator<'a, 'a>,
+}
+impl<'a> IteratorWrapper<'a> {
+    fn next_column(&mut self) -> PyResult<Column> {
+        Python::with_gil(|py| {
+            let raw_col = self
+                .column_iterator
+                .next()
+                .ok_or_else(|| PyErr::new::<pyo3::exceptions::PyStopIteration, _>(""))?
+                .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+            let val = PyDeserializedValue::deserialize_py(raw_col.spec.typ(), raw_col.slice, py)
+                .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+            let column_name = PyString::new(py, raw_col.spec.name()).unbind();
+            let value = val.into_pyobject(py).expect("Can't fail").unbind();
+
+            Ok(Column { column_name, value })
+        })
+    }
+
+    fn next_row(&mut self) -> PyResult<()> {
+        let column_iterator = self
+            .row_iterator
+            .next()
+            .ok_or_else(|| PyErr::new::<pyo3::exceptions::PyStopIteration, _>(""))?
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+        self.column_iterator = column_iterator;
+        Ok(())
+    }
+}
+
+// NOTE: In method attach_to_cart we cannot move out cart so We cannot use into_inner inside of it.
+// Fields in DeserializedMetadataAndRawRows are private We have to think
+// If it would be better to use method into_inner and construct new RawMetadata struct
+// or to implement some public method in rust driver that would give access to raw_rows.
+struct RawMetadata {
+    metadata: ResultMetadataHolder,
+    rows_count: usize,
+    raw_rows: Bytes,
+}
+
+impl RawMetadata {
+    fn from_inner(data: DeserializedMetadataAndRawRows) -> RawMetadata {
+        let (metadata, rows_count, raw_rows) = data.into_inner();
+
+        RawMetadata {
+            metadata,
+            rows_count,
+            raw_rows,
+        }
+    }
+}
+
+struct RawMetadataCart(Box<RawMetadata>);
+
+impl Deref for RawMetadataCart {
+    type Target = RawMetadata;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+unsafe impl StableDeref for RawMetadataCart {}
+
+#[pyclass(name = "ColumnIterator")]
+pub struct RowColumnIterator {
+    yoked: Yoke<IteratorWrapper<'static>, RawMetadataCart>,
+}
```

Comment:
:question: Why is this called "RowColumnIterator"? What's a "RowColumn"? How do I parse this?